File: frontend/src/api/season.ts
/**
 * Return the current NFL season year.
 *
 * The NFL season starts in September, so before September we still consider
 * the previous calendar year to be the "current" season (e.g. in March 2026
 * the active season is 2025).
 */
export function getCurrentNflSeason(): number {
  const now = new Date();
  const year = now.getFullYear();
  const month = now.getMonth(); // 0-indexed: 0=Jan, 8=Sep
  return month < 8 ? year - 1 : year;
}

File: backend/app/core/season.py
from datetime import UTC, datetime


def get_current_nfl_season() -> int:
    """Return the current NFL season year.

    The NFL season starts in September, so before September we still
    consider the previous calendar year to be the current season
    (e.g. March 2026 → 2025).
    """
    now = datetime.now(UTC)
    return now.year if now.month >= 9 else now.year - 1
